### DISCUSSION TITLE
Actions for unix equivalent of while do loop

### DISCUSSION THREAD

**Discussion by haifrancis** [community/community#52061] [Actions] [answered]
### Select Topic Area

Question

### Body

We have a set of files listed in a manifest file. We need to loop through each line of the manifest and verify if the listed file is present at the directory. Currently we do this using a simple unix shell script to loop through the manifest and a file check like "if ! [ -f $filename ]". Is there any alternate way of achieving this using actions as we would like to keep this generic across OS.
We need to be able to verify existence of all files in the manifest and return the status.

**Comment by meppu** [comment, marked as answer]
Easiest way to solve is using bash for windows too. As you can see in [Shell Options](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsshell) `bash` is supported for all platforms. So you can use it like this:

```yaml
steps:
  - name: Print path
    run: echo $PATH
    shell: bash
```
